oracle_cardano/processor/tx_processors/success/bridging_requested_processor_skyline.go

```diff
@@ -0,0 +1,345 @@
+	feeSum := uint64(0)
+	foundAUtxoValueBelowMinimumValue := false
+	foundAnInvalidReceiverAddr := false
+
+	exchangeRate, err := GetExchangeRate(metadata.DestinationChainID)
+	if err != nil {
+		return err
+	}
+
+	for _, receiver := range metadata.Transactions {
+		receiverAddr := strings.Join(receiver.Address, "")
+
+		var totalAmountToBridge uint64
+
+		if cardanoDestConfig != nil {
+			if receiver.Additional == nil {
+				if !cardanotx.IsValidOutputAddress(receiverAddr, cardanoDestConfig.NetworkID) {
+					foundAnInvalidReceiverAddr = true
+
+					break
+				}
+
+				if receiverAddr == cardanoDestConfig.BridgingAddresses.FeeAddress {
+					feeSum += receiver.Amount
+
+					continue
+				} else {
+					foundAUtxoValueBelowMinimumValue = true
+
+					break
+				}
+			}
+
+			if receiver.IsNativeToken {
+				// amount_to_bridge must be >= minUtxoAmount on destination
+				if receiver.Amount < cardanoDestConfig.UtxoMinAmount {
+					foundAUtxoValueBelowMinimumValue = true
+
+					break
+				}
+
+				totalAmountToBridge = receiver.Amount
+			} else {
+				// amount_to_bridge must be >= minUtxoAmount on source
+				if receiver.Amount < cardanoSrcConfig.UtxoMinAmount {
+					foundAUtxoValueBelowMinimumValue = true
+
+					break
+				}
+
+				if receiver.Additional.DestinationCurrencyAmount < cardanoDestConfig.UtxoMinAmount {
+					foundAUtxoValueBelowMinimumValue = true
+
+					break
+				}
+
+				totalAmountToBridge = receiver.Amount
+			}
+
+			if uint64(float64(receiver.Additional.DestinationCurrencyAmount)*exchangeRate) !=
+				receiver.Additional.SourceCurrencyAmount {
+				return fmt.Errorf("found an exchange rate error in metadata: %v", metadata)
+			}
+
+			if !cardanotx.IsValidOutputAddress(receiverAddr, cardanoDestConfig.NetworkID) {
+				foundAnInvalidReceiverAddr = true
+
+				break
+			}
+
+			receiverAmountSum.Add(receiverAmountSum, new(big.Int).SetUint64(totalAmountToBridge))
+		} else if ethDestConfig != nil {
+			if !goEthCommon.IsHexAddress(receiverAddr) {
+				foundAnInvalidReceiverAddr = true
+
+				break
+			}
+
+			if receiverAddr == common.EthZeroAddr {
+				feeSum += receiver.Amount
+			} else {
+				receiverAmountSum.Add(receiverAmountSum, new(big.Int).SetUint64(receiver.Amount))
+			}
+		}
+	}
+
+	if foundAUtxoValueBelowMinimumValue {
+		return fmt.Errorf("found a utxo value below minimum value in metadata receivers: %v", metadata)
+	}
+
+	if foundAnInvalidReceiverAddr {
+		return fmt.Errorf("found an invalid receiver addr in metadata: %v", metadata)
+	}
+
+	// update fee amount if needed with sum of fee address receivers
+	feeAmount := metadata.FeeAmount.DestinationCurrencyAmount + feeSum
+	receiverAmountSum.Add(receiverAmountSum, new(big.Int).SetUint64(feeAmount))
+
+	if cardanoDestConfig != nil && feeAmount < cardanoDestConfig.MinFeeForBridging {
+		return fmt.Errorf("bridging fee in metadata receivers is less than minimum: %v", metadata)
+	}
+
+	if receiverAmountSum.Cmp(new(big.Int).SetUint64(multisigUtxo.Amount)) != 0 {
```

Comment:
here, you should check if receiverAmountSum == multisigUtxo.Amount and if receiverWrappedTokenAmountSum == multisifUtxo.NativeTokens[corresponding token on source].Amount